src/File.php

```diff
@@ -485,6 +485,16 @@ public function canDelete($member = null)
             ->canDelete($this->ID, $member);
     }
 
+    /**
+     * Check if the File is viewable to a public user i.e. someone who is not logged in to the CMS
+     */
+    public function canViewAnonymous(): bool
```

Comment:
This should possibly be on DataObject instead

Reply:
Arguably we should have written the `canView` methods to accept a value that means "Anonymous". Unfortunately, any falsy value gets converted to the current user which is annoying.

Reply:
If we call this method on every file every time you display a list of files, it might end up being very expansive.

You might be able to do something like get the permission of the parent folder and have the front-end or graphql decide if if the file is restricted or not based on each file's CanViewType property.

Reply:
hmm yeah not too keen to build a separate canView() set of logic on the front end.  If it does turn out to be too expensive then I'd way rather prefer to look at optimising things in the backend